plugins/migration/bin/rename.js

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env node
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A script that automatically upgrades pieces of the Blockly API
+ *     that have been renamed by modifying developers' local files.
+ */
+'use strict';
+
+import {createAndAddSubCommand} from './command.js';
+import fetch from 'node-fetch';
+import glob from 'glob';
+import {readFile, writeFile} from 'fs';
+import semver from 'semver';
+import JSON5 from 'json5';
+
+const DATABASE_URL = `https://raw.githubusercontent.com/google/blockly/develop/scripts/migration/renamings.json5`;
+
+/**
+ * The version associated with renamings that have not been released yet.
+ * @const {string}
+ */
+const DEV_VERSION = 'develop';
+
+createAndAddSubCommand(
```

Comment:
Is there a way that we could make this module side-effect free, and instead export the subcommand?  (That would also give `migrate.js` a bit more purpose.)

Reply:
I was trying to make sure that `ROOT_COMMAND` could only have subcommands with the given form, by making it so that the only external access to it was via `createAndAddSubcommand` and `parseAndRunMigrations`. IE no `addSubCommand` function was available.

I could also create a subclass of `Command` and use type annotations to "ensure" those are the only things that get added to the `ROOT_COMMAND`... but that seems like overkill. I'll just add an `addSubCommand` function to remove side effects, and hope nobody decides to break the pattern in the future.